bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/VehicleHandler.java

```diff
@@ -0,0 +1,796 @@
+            if (content != null) {
+                lastTripCache = Optional.of(content);
+                try {
+                    LastTripContainer lt = Converter.getGson().fromJson(content, LastTripContainer.class);
+                    if (lt != null) {
+                        LastTrip trip = lt.lastTrip;
+                        if (trip != null) {
+                            updateLastTrip(trip);
+                        }
+                    }
+                } catch (JsonSyntaxException jse) {
+                    logger.debug("{}", jse.getMessage());
+                }
+            }
+            removeCallback(this);
+        }
+
+        /**
+         * Store Error Report in cache variable. Via Fingerprint Channel error is logged and Issue can be raised
+         */
+        @Override
+        public void onError(NetworkError error) {
+            logger.debug("{}", error.toString());
+            lastTripCache = Optional.of(Converter.getGson().toJson(error));
+            removeCallback(this);
+        }
+    }
+
+    /**
+     * The VehicleStatus is supported by all Vehicle Types so it's used to reflect the Thing Status
+     */
+    public class VehicleStatusCallback implements StringResponseCallback {
+        private ThingStatus thingStatus = ThingStatus.UNKNOWN;
+
+        @Override
+        public void onResponse(@Nullable String content) {
+            if (content != null) {
+                updateStatus(ThingStatus.ONLINE);
+                vehicleStatusCache = Optional.of(content);
+                try {
+                    VehicleStatusContainer status = Converter.getGson().fromJson(content, VehicleStatusContainer.class);
+                    if (status != null) {
+                        VehicleStatus vStatus = status.vehicleStatus;
+                        if (vStatus == null) {
+                            return;
+                        }
+                        updateVehicleStatus(vStatus);
+                        updateCheckControls(vStatus.checkControlMessages);
+                        updateServices(vStatus.cbsData);
+                        updatePosition(vStatus.position);
+                    }
+                } catch (JsonSyntaxException jse) {
+                    logger.debug("{}", jse.getMessage());
+                }
+            }
+            removeCallback(this);
+        }
+
+        @Override
+        public void onError(NetworkError error) {
+            logger.debug("{}", error.toString());
+            if (error.status == 404) {
+                legacyMode = true;
+                logger.debug("VehicleStatus not found - switch to legacy API");
+                proxy.get().requestLegacyVehcileStatus(configuration.get(), oldVehicleStatusCallback);
+            }
+            vehicleStatusCache = Optional.of(Converter.getGson().toJson(error));
+            setThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error.reason);
```

Comment:
I get a compile-error here - method 'setThingStatus' being unknown.
fixed in https://github.com/weymann/openhab-addons/pull/12

Reply:
Last 3 builds are working fine - perhaps rebase with the newest baseline missing with the latest review changes?

Reply:
in https://github.com/weymann/openhab-addons/commit/5a568b041a01f7b8cd6e1290a8d969c40369480a you did delete method 'setThingStatus' which caused the compile-error (being resoleved in https://github.com/weymann/openhab-addons/commit/3eb30f7cce7556b0109707518ab4c304dc39fe18).

Reply:
ok, so resolved